File: Old JAVA/Enumerise/NutrientAnalysis.java
package Enumerise;

import java.util.HashMap;
import java.util.Map;

public enum NutrientAnalysis implements Nutrient {
NaK("Rapport K/NA",0,"", "KNA"),
PCa("Rapport phosphocalcique",1,"", "CAP"),
o6o3("Rapport omega 6/omega3",2,"", "O6O3"),

ZnCu("Rapport Zn/Cu",3,"", "ZNCU"),
nonOsPhos("Phosphore non osseux",7,"%", "nonOsPhos"),
nonOsProt("Proteine non osseuse",8,"%", "nonOsProt"),
nonOsPP("Ratio Prot/phos non osseux",9,"", "nonOsPP"),
PhosphProt("Rapport Prot??ines/Phosphore",4,"", "PROTP"),
MethCys("Methionine+cyst??ine",5,"g", "METHCYS"),
PhenTyr("Ph??nylalanine+tyrosine",6,"g", "PHENTYR")
			;
			


		private String name = "";
		private String unite = "";
		private int coef = 0;
		private String Label="";
	

		//Constructeur
	NutrientAnalysis(String name, int coef, String unite, String label){
		this.name = name;
		this.coef=coef;
		this.unite=unite;
		this.Label=label;
		}
	public static NutrientAnalysis getByCoef(int i){
	
		return (NutrientAnalysis) map.get(i);
	}
	 private static Map map = new HashMap<>();
	static {
        for (NutrientAnalysis pageType : NutrientAnalysis.values()) {
            map.put(pageType.coef, pageType);
        }
    }
	
	
	public String getUnite(){
	return unite;
	}
public MainNutrientEnum getMNE() {
	return MainNutrientEnum.ANA;
}
		public String nameToString(){
		return name;
		}
		public int getCoef(){
			  return coef;
			}
		public static int size(){
			  return 7;
			}
		 public String getLabel() {
			return Label;
		}
		@Override
		public UnitEnum getUe() {
			// TODO Auto-generated method stub
			return UnitEnum.NO;
		}
		
		}

File: composeApp/src/commonMain/kotlin/fr/vetbrain/vetnutri_mp/Enumer/NutrientAnalysis.kt
package fr.vetbrain.vetnutri_mp.Enumer


enum class NutrientAnalysis(
    val displayName: String,
    override val coef: Int,
    override val unite: String,
    override val label: String

) : Nutrient {
    NaK("Rapport K/NA", 0, "", "KNA"),
    PCa("Rapport Ca/P", 1, "", "CAP"),
    o6o3("Rapport O6/O3", 2, "", "O6O3"),
    ZnCu("Rapport Zn/Cu", 3, "", "ZNCU"),
    nonOsPhos("Phosphore non osseux", 7, "%", "nonOsPhos"),
    nonOsProt("Proteine non osseuse", 8, "%", "nonOsProt"),
    nonOsPP("Ratio Prot/phos non osseux", 9, "", "nonOsPP"),
    PhosphProt("Rapport Prot/Phos", 4, "", "PROTP"),
    MethCys("Méthionine+cystéine", 5, "g", "METHCYS"),
    PhenTyr("Phénylalanine+tyrosine", 6, "g", "PHENTYR");

    companion object {
        private val coefMap by lazy { entries.associateBy { it.coef } }
        private val labelMap by lazy { entries.associateBy { it.label } }

        fun getByCoef(coef: Int) = coefMap[coef]
        fun getByLabel(label: String) = labelMap[label]
        fun isByLabel(label: String) = label in labelMap
        fun size() = entries.size
    }


    override fun getMNE() = MainNutrientEnum.ANA
    override val ue = UnitEnum.NO
    // Redondance supprimée grâce aux propriétés Kotlin
}